src/links/links.service.ts

```diff
@@ -15,7 +15,7 @@ export class LinksService {
 
   async findOne(id: Link['id']): Promise<Link | null> {
```

Comment:
**issue**: Add a null check for `link` to prevent runtime errors if `link` is not found.